protocol/messenger_communities.go

```diff
@@ -3567,28 +3589,25 @@ func (m *Messenger) UpdateCommunityTokenState(contractAddress string, deployStat
 	return m.communitiesManager.UpdateCommunityTokenState(contractAddress, deployState)
 }
 
-// UpdateCommunityEncryption takes a community and encrypts / decrypts the community
-// based on TokenPermission. Community is republished along with any keys if needed.
+// UpdateCommunityEncryption takes a communityID string and an encryption state, then finds the community and
+// encrypts / decrypts the community. Community is republished along with any keys if needed.
 //
 // Note: This function cannot decrypt previously encrypted messages, and it cannot encrypt previous unencrypted messages.
 // This functionality introduces some race conditions:
 //   - community description is processed by members before the receiving the key exchange messages
 //   - members maybe sending encrypted messages after the community description is updated and a new member joins
-func (m *Messenger) UpdateCommunityEncryption(community *communities.Community) (*MessengerResponse, error) {
+func (m *Messenger) UpdateCommunityEncryption(community *communities.Community, useEncryption bool) (*MessengerResponse, error) {
 	if community == nil {
 		return nil, errors.New("community is nil")
 	}
 
-	becomeMemberPermissions := community.TokenPermissionsByType(protobuf.CommunityTokenPermission_BECOME_MEMBER)
-	isEncrypted := len(becomeMemberPermissions) > 0
-
```

Comment:
As noted above I've removed external concerns from `Community` encryption management as the role of this function should be to update a `Community`'s encryption state based ONLY on a boolean, this allows all calling code to set their conditions independent of other state concerns.